src/gluonts/ev/interface.py

```diff
@@ -0,0 +1,191 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+from typing import ChainMap, Tuple, Optional, Iterable, Union
+from dataclasses import dataclass
+
+import numpy as np
+import pandas as pd
+
+from gluonts.dataset import DataEntry
+from gluonts.dataset.split import TestData
+from gluonts.ext.naive_2 import naive_2
+from gluonts.time_feature.seasonality import get_seasonality
+from gluonts.model.forecast import Forecast
+from gluonts.ev import seasonal_error
+
+
+@dataclass
+class ForecastDict:
+    """
+    Wrapper around ``Forecast`` objects, that adds ``__getitem__`` and
+    a batch dimension to returned arrays, for compatibility with ``gluonts.ev``.
+    """
+
+    forecast: Forecast
+
+    def __getitem__(self, name):
+        if name == "mean":
+            value = self.forecast.mean
+        elif name == "median":
+            value = self.forecast.median
+        else:
+            value = self.forecast.quantile(name)
+
+        return np.expand_dims(value, axis=0)
+
+
+def _get_data_batch(
+    input_: DataEntry, label: DataEntry, forecast: Forecast
+) -> ChainMap:
+    forecast_dict = ForecastDict(forecast)
+
+    freq = forecast.start_date.freqstr
+    seasonality = get_seasonality(freq=freq)
+
+    other_data = {
+        "label": np.array([label["target"]]),
+        "seasonal_error": np.array(
+            [seasonal_error(input_["target"], seasonality=seasonality)]
+        ),
+        "naive_2": np.array(
+            [naive_2(input_["target"], len(label["target"]), freq=freq)]
+        ),
+    }
+
+    return ChainMap(other_data, forecast_dict)
+
+
+def evaluate(
+    metrics, test_data: TestData, forecasts: Iterable[Forecast], axis=None
+) -> Tuple[dict, Optional[list]]:
+    """
+    Evaluate ``metrics`` by comparing ``forecasts`` to ``test_data``.
+
+    The optional ``axis`` arguments controls aggregation of the metrics:
+    - ``None`` (default) aggregates across all dimensions
+    - ``0`` aggregates across the batch dimension, giving one metric value per forecast horizon
+    - ``1`` aggregates across the time dimension, giving one metric value per batch item
+
+    Return results as a dictionary, plus an optional index (list).
```

Comment:
What is the index for?

Reply:
It's mainly for the downstream `evaluate_pd` (see below, name is tentative). The thing is, as long as you provide metric values as arrays, it's hard to read through them if you don't know what the axes are. For example, which `item_id` does this "0.12345" MAPE value refers to?

This is all up for discussion, but I think that an "interface" at this level should provide a more readable structure than plain arrays.